src/datanode/src/region_server.rs

```diff
@@ -827,7 +832,10 @@ impl TableProvider for DummyTableProvider {
         limit: Option<usize>,
     ) -> DfResult<Arc<dyn DfPhysicalPlan>> {
         let mut request = self.scan_request.lock().unwrap().clone();
-        request.projection = projection.cloned();
+        request.projection = match projection {
+            Some(x) if !x.is_empty() => Some(x.clone()),
+            _ => None,
+        };
```

Comment:
What if we use `request.projection = projection.cloned();` here? Though `Some(vec![])` seems meaningless.